localstack-core/localstack/services/sns/provider.py

```diff
@@ -943,6 +958,11 @@ def validate_message_attributes(message_attributes: MessageAttributeMap) -> None
             raise InvalidParameterValueException(
                 f"The message attribute '{attr_name}' has an invalid message attribute type, the set of supported type prefixes is Binary, Number, and String."
             )
+        if not any(attr_value.endswith("Value") for attr_value in attr):
+            raise InvalidParameterValueException(
+                f"The message attribute '{attr_name}' must contain non-empty message attribute value for message attribute type '{data_type}'."
+            )
```

Comment:
Question: It appears that only `StringValue` and `BinaryValue` are supported in the message attributes. Is there a reason to allow `<Any>Value`?

Reply:
Good point... I've tried to run a test with `BadValue` but I get a serialization error from the client 😭 
```python
prefix = 'MessageAttributes.entry.1.Value'

    def _serialize_type_structure(self, serialized, value, shape, prefix=''):
        members = shape.members
        for key, value in value.items():
>           member_shape = members[key]
E           KeyError: 'BadValue'
```

I'd need to get the raw request to validate... 

I guess then it will be a matter of raising the right exception. I think if I accept `<Any>Value`, then it will raise `InvalidParameterValueException(f"The message attribute '{attr_name}' with type '{data_type}' must use field '{value_key_data_type}'.")`. Not sure if it's the right behavior but at least we raise something. I guess I won't fight it too much, but it's very good point... 🤔